### PR TITLE
feat: 'citation' is a reserved name & disallow '__' name prefixes

### PR DESCRIPTION
**Pull request by danieljhegeman**
## Reason for Change

- #519

## Changes

- Add 'citation' as a reserved_column for the uns component
- Do not allow columns/keys to be prefaced with '__' for:
  - uns
  - obs
  - obs.var
  - obs.raw.var

## Testing

- Either list QA steps or reasoning you feel QA is unnecessary
- Reminder For CLI changes: upon merge, contact Lattice for final sign-off. Do not release a new cellxgene-schema 
version to PyPI without explicit QA + sign-off from Lattice on all functional CLI changes. They may install the package
version at HEAD of main with 
```
pip install git+https://github.com/chanzuckerberg/single-cell-curation/@main#subdirectory=cellxgene_schema_cli
```

## Notes for Reviewer

### CODE REVIEW

File: cellxgene_schema_cli/cellxgene_schema/validate.py [hunk 1]
```diff
@@ -677,6 +680,12 @@ def _validate_dataframe(self, df_name: str):
                         self.warnings.append(column_def["warning_message"])
                     self._validate_column(column, column_name, df_name, column_def)
 
+        # Validate user-provided column names
+        for column in df.columns:
+            if re.match(r"__", column):  # re.match matches strictly from beginning of str
+                self.errors.append(f"The column '{column}' in dataframe '{df_name}' is not valid; columns must not "
+                                   f"have a '__' prefix.")
+
```
Comment (danieljhegeman) [inline cellxgene_schema_cli/cellxgene_schema/validate.py:688]: right now this means `var`, `raw.var`, and `obs`, since they are the 3 components of type `dataframe` in the schema 

see [conversation](https://czi-sci.slack.com/archives/C03PXFMM7ST/p1695255977203039?thread_ts=1694818587.348629&cid=C03PXFMM7ST)

File: cellxgene_schema_cli/cellxgene_schema/validate.py [hunk 2]
```diff
@@ -580,10 +580,8 @@ def _validate_dict(self, dictionary: dict, dict_name: str, dict_def: dict):
 
         for key, value_def in dict_def["keys"].items():
             logger.debug(f"Validating uns dict for key: {key}")
-            if key not in dictionary:
-                if "required" in value_def:
-                    self.errors.append(f"'{key}' in '{dict_name}' is not present.")
-                continue
+            if key not in dictionary and "required" in value_def:
+                self.errors.append(f"'{key}' in '{dict_name}' is not present.")
```
Comment (danieljhegeman) [inline cellxgene_schema_cli/cellxgene_schema/validate.py:584]: unrelated lint fix

**Review by nayib-jose-gloria — COMMENTED** [review]
need to fix the tests--maybe worth noting that 'citation' will be a unique case where the data portal will be annotating the field rather than the CLI. This is because it requires context about the collection its uploaded to (namely the publication DOI)
